src/lib/checkpoint/cp_mgr.cpp

```diff
@@ -205,6 +214,54 @@ void CPManager::cleanup_cp(CP* cp) {
     }
 }
 
+//////////////////////////////////////// CP Guard class ////////////////////////////////////////////
+CPGuard::CPGuard() {
+    if (t_cp_stack.empty()) {
+        // First CP in this thread stack.
+        m_cp = hs()->cp_mgr().cp_io_enter();
```

Comment:
m_cp can be nullptr in some case, is it expected?

Reply:
yes you are right. I can remove the assert there. But in reality cp_io_enter() returns nullptr can happen only after CPManager::shutdown has been called and so that should not happen, but if it happens is it better to fire off the assert or not?

Reply:
shutdown should not be triggered if any cp is in progress, right? If so, we probably should keep the assert.